src/haptics.cpp

```diff
@@ -21,42 +56,37 @@ init_haptics() {
  */
 void actuate_motor(int motor_pin, int duration, double percent_motor) {
   digitalWrite(MOTOR_PINS[motor_pin] + EN_OFFSET, HIGH);  // enable
-  int frequency = 3;
+  int half_period = 3;
   if (percent_motor > 0.85) {
     Serial.println("Motor percentage above 85%. Using 85%");
     percent_motor = 0.85;
   }
-  int numCycles = duration / (frequency * 2);
+  percent_motor = 1;
+  int numCycles = duration / (half_period * 2);
   for (int j = 0; j < numCycles; j++) {  // do J_MAX * TODAL_DELAY = 16 * 6 = 96ms of vibration
     analogWrite(MOTOR_PINS[motor_pin], 512 + percent_motor * 512);
-    delay(frequency);
+    delayMicroseconds(half_period * 1000);
     analogWrite(MOTOR_PINS[motor_pin], 512 - percent_motor * 512);
-    delay(frequency);
+    delayMicroseconds(half_period * 1000);
   }
+  analogWrite(MOTOR_PINS[motor_pin], 512);
   digitalWrite(MOTOR_PINS[motor_pin] + EN_OFFSET, LOW);  // disable
 }
 
-/* Repeats vibration mode for given duration
- * @param motor_pin	specified motor to vibrate
- * @param duration_on	length time of vibration in ms
- * @param percent_motor percent of haptic vibration capacity to vibrate at,
- * 			not to exceed 0.85
- * @param duration_off length of delay time between vibration cycle
- */
 
-void repeat_motor(int motor_pin, int repeat, int duration_on, int duration_off, double percent_motor) {
-     for(int i = 0; i  < repeat; i++){
-	 actuate_motor(motor_pin, duration_on, percent_motor);
-	 delay(duration_off);
-      }
+void run_haptics(pulse* p, int num_pulses, int num_cycles/*, int motors[]*/) {
+  for (int i = 0; i < num_cycles; i++) { // number of cycles to do total pattern
+    for (int j = 0; j < num_pulses; j++) {
+      actuate_hz(0, p[j].dur, 0.85, p[j].freq);
```

Comment:
thoughts on making the intensity a variable to this function?

Reply:
due to weird motor driver stuff, the sine wave of the voltage across the motor appears to get weird at lower intensities, which is why I think 0.85 might be a good default. If other intensities work too though I'm open